### ISSUE TITLE
feat(radtran): SWIR surface albedo / BRDF and TIR emissivity models

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/plumax#98]
## Problem / Request
`radtran.forward_*` and `coupled/rtm.py::radiance_response` model the plume as `L = L₀ · exp(−Δτ)` with a normalised background — the surface enters only as a per-band constant folded into `L₀`. Real SWIR scenes have wavelength-dependent albedo (and sun-glint / vegetation BRDF) that aliases into the methane band, and TIR instruments need an emissivity + surface-temperature term. Both surface models are ☐ (`radtran.surface_swir`, `radtran.surface_tir`); the design keeps Lambertian for v1 with a BRDF registry as v1.5/v2.

## User Story
> As a user retrieving over bright desert and dark water in the same scene, I want a per-pixel surface model in the forward (Lambertian albedo spectrum v1, BRDF-by-land-use later) and a TIR emissivity path, so surface structure doesn't masquerade as methane.

## Proposed API
```python
# src/plumax/radtran/surface_swir.py
@dataclass(frozen=True)
class LambertianSurface:  albedo: Float[Array, "... B"]        # per band, retrievable (linear or 2nd-order polynomial in λ)
class BRDF(Protocol):     def __call__(self, sza, vza, raa, wavelength) -> jax.Array: ...
BRDF_REGISTRY: dict[str, BRDF]   # "lambertian", "cox_munk", "ross_li" (v2)

# src/plumax/radtran/surface_tir.py
@dataclass(frozen=True)
class TIRSurface: emissivity: Float[Array, "... B"]; temperature_k: Float[Array, "..."]
def tir_radiance(surface: TIRSurface, atmosphere: Background, tau) -> jax.Array: ...   # ε B(T_s) τ + (1 − ε) L↓ τ + L↑
```

## Implementation Steps
- [ ] Add `surface_swir.py`; thread `LambertianSurface` through `forward_nonlinear*` as a multiplicative per-band term with a polynomial-in-λ parameterisation that the OE loop can retrieve as nuisance state
- [ ] Add `surface_tir.py` with the Planck-based emission path and its Jacobians
- [ ] `RadianceObservationModel` grows optional `surface` state; `coupled/rtm.py` passes it through

## Definition of Done
- [ ] With unit albedo / unit emissivity the forward is bit-identical to today
- [ ] Synthetic scene with a 10 % albedo slope: OE with the surface state recovers the column unbiased; without it, biased
- [ ] `make test && make lint && make typecheck` green

## Testing
- [ ] `tests/radtran/test_surface_swir.py::test_unit_albedo_identity`
- [ ] `tests/radtran/test_surface_swir.py::test_albedo_slope_bias_removed` (`slow`)

## Relationships
- Parent (theme epic): #73
- Related: optimal-estimation retrieval (surface as nuisance state)
